lib/libutils/ext/trace.c

```diff
@@ -91,28 +117,32 @@ void trace_printf(const char *function, int line, int level, bool level_ok,
 		return;
 	boffs += res;
 
-	/* Print the Thread ID */
-	if (level_ok && !(BIT(level) & CFG_MSG_LONG_PREFIX_MASK))
-		thread_id = -1;
-	else
-		thread_id = trace_ext_get_thread_id();
-
-	res = print_thread_id(buf + boffs, sizeof(buf) - boffs, thread_id);
-
-	if (res < 0)
-		return;
-	boffs += res;
-
-	/* Print the function and line */
-	if (level_ok && !(BIT(level) & CFG_MSG_LONG_PREFIX_MASK))
-		function = NULL;
+	if (level_ok && (BIT(level) & CFG_MSG_LONG_PREFIX_MASK)) {
+		/* Print the core ID if in atomic context  */
+		res = print_core_id(buf + boffs, sizeof(buf) - boffs);
+		if (res < 0)
+			return;
+		boffs += res;
 
-	if (function) {
-		res = snprintk(buf + boffs, sizeof(buf) - boffs, "%s:%d ",
-			       function, line);
+		/* Print the Thread ID */
+		res = print_thread_id(buf + boffs, sizeof(buf) - boffs);
 		if (res < 0)
 			return;
 		boffs += res;
+
+		if (function) {
+			res = snprintk(buf + boffs, sizeof(buf) - boffs, "%s:%d ",
+				       function, line);
+			if (res < 0)
+				return;
+			boffs += res;
+		}
+	} else {
+		/* Add space after location info */
+		if (boffs >= sizeof(buf))
+		    return;
+		buf[boffs++] = ' ';
+		buf[boffs] = 0;
```

Comment:
If `boffs == (MAX_PRINT_SIZE - 1)` when entering the `else {}`, this line will overflow

Reply:
Good catch. Thank you. I updated the whole PR, because problem code was added in fixup and I decided that fixup for fixup is overkill :)